codebasin/__main__.py

```diff
@@ -144,12 +145,25 @@ def main():
 
     args = parser.parse_args()
 
-    stdout_log = logging.StreamHandler(sys.stdout)
-    stdout_log.setFormatter(logging.Formatter("[%(levelname)-8s] %(message)s"))
-    logging.getLogger("codebasin").addHandler(stdout_log)
-    logging.getLogger("codebasin").setLevel(
-        max(1, logging.WARNING - 10 * (args.verbose - args.quiet)),
-    )
+    # Configure logging such that:
+    # - All messages are written to a log file
+    # - Only errors are written to the terminal by default
+    # - Messages written to terminal are based on -q and -v flags
+    formatter = logging.Formatter("[%(levelname)-8s] %(message)s")
+    log.setLevel(logging.DEBUG)
+
+    file_handler = logging.FileHandler("cbi.log", mode="w")
+    file_handler.setLevel(logging.INFO)
+    file_handler.setFormatter(formatter)
+    log.addHandler(file_handler)
+    log_path = os.path.realpath("cbi.log")
+    print(f"Log file created at {log_path}")
```

Comment:
Can you not just log this _after_ you've set up the `stdout` handler?

Reply:
Hm.  This is a tricky one, I think.

Written this way, the message is part of CBI's output.  It's always printed to the terminal, regardless of how many `-q` or `-v` flags the user passed.  It's (hopefully) very difficult for a user to miss this message, but it won't be logged.

If we wrote this to the log using `log.info`, it wouldn't show up to `stdout` by default.  A user would have to run with `-vv` to see it, and that would also turn on all the warnings and other information messages, too.

One possible, but slightly hacky, solution would be to write it like this:
```python
log.addHandler(stdout_handler)
log.info(f"Log file created at {log_path}")
stdout_handler.setLevel(log_level)
```

Logging before we call `setLevel` means the message displays unconditionally, but also gets written to the log.  There's probably a smarter way to set up different logging rules for `__main__.py`, but that might be overkill if we only care about this one message.

Is that too much of a hack, or do you think it's okay?

Reply:
I think the reasoning behind printing is good enough for me as you described it - I think I just have a natural aversion to `print` as "not very professional UX".

I'll mark as resolved, but if you don't mind dropping a comment line there explaining why `print` specifically is used.